core/src/main/java/io/temporal/samples/hello/CommandProcessor.java

```diff
@@ -0,0 +1,137 @@
+package io.temporal.samples.hello;
+
+import io.temporal.activity.ActivityInterface;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.api.enums.v1.WorkflowIdReusePolicy;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.client.WorkflowStub;
+import io.temporal.serviceclient.WorkflowServiceStubs;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import io.temporal.workflow.CompletablePromise;
+import io.temporal.workflow.UpdateMethod;
+import io.temporal.workflow.Workflow;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.concurrent.CompletableFuture;
+
+public class CommandProcessor {
+  static final String TASK_QUEUE = "MyTaskQueue";
+  static final String WORKFLOW_ID = "MyWorkflowId";
+
+  @WorkflowInterface
+  public interface CommandProcessorWorkflow {
+    @WorkflowMethod
+    String startProcessing();
+
+    @UpdateMethod
+    String submitCommand(int command);
+  }
+
+  @ActivityInterface
+  public interface MyActivities {
+    String processCommand(int command);
+  }
+
+  public static class CommandProcessorWorkflowImpl implements CommandProcessorWorkflow {
+
+    private static class QueuedCommand {
+      public int command;
+      public CompletablePromise<String> promise;
+
+      public QueuedCommand(int command, CompletablePromise<String> promise) {
+        this.command = command;
+        this.promise = promise;
+      }
+    }
+
+    private ArrayList<QueuedCommand> commandQueue;
```

Comment:
`ArrayList` works in this case, but prefer `WorkflowQueue` instead. It provides a more appropriate API, inspired from "Blocking Queue", but Workflow-safe (it internally use `Workflow.await()` to implement blocking ops).

Reply:
Thanks! Yes @Quinn-With-Two-Ns pointed that out to me also.